packages/fleet/lib/db/migrations/20241102115903_initial_fleet_models.ts

```diff
@@ -0,0 +1,57 @@
+import type { Knex } from 'knex';
+import { NODES_TABLE } from '../../models/nodes.js';
+import { DEPLOYMENTS_TABLE } from '../../models/deployments.js';
+
+export async function up(knex: Knex): Promise<void> {
+    await knex.transaction(async (trx) => {
+        await trx.raw(`
+            CREATE TABLE ${DEPLOYMENTS_TABLE} (
+                commit_id char(40) PRIMARY KEY,
+                created_at timestamp with time zone NOT NULL DEFAULT CURRENT_TIMESTAMP,
+                superseded_at timestamp with time zone
+            );
+        `);
+        await trx.raw(`
+            CREATE INDEX idx_${DEPLOYMENTS_TABLE}_active ON ${DEPLOYMENTS_TABLE}(superseded_at) WHERE superseded_at IS NULL;
+        `);
+        await trx.raw(`
+            CREATE TYPE node_states AS ENUM (
+                'PENDING',
+                'STARTING',
+                'RUNNING',
+                'OUTDATED',
+                'FINISHING',
+                'IDLE',
+                'TERMINATED',
+                'ERROR'
+            );
+        `);
+        await trx.raw(`
+            CREATE TABLE ${NODES_TABLE} (
+                id SERIAL PRIMARY KEY,
+                routing_id varchar(255) NOT NULL,
+                deployment_id char(40) NOT NULL REFERENCES deployments(commit_id) ON DELETE CASCADE,
```

Comment:
Need a unique index for those two no?

Reply:
I now remember why I didn't make it unique. Because any node can end up in ERROR state and therefore several entry can have the same routingId/deploymentId. Ex:
1. Node is created as PENDING
2. Request to Render to start the new runner fails for some reason. Node is set to ERROR
3. A new node with same routingID and deploymentId is created as PENDING

Reply:
I see, could be nice to have this clarified somewhere in the models 👍🏻